File: test/credible/17.benchmark/wrk/tps.lua
-- example HTTP POST script which demonstrates setting the
-- HTTP method, body, and adding a header

counter = 0

request = function()
    totalCounter = 100000000 + counter
   path = "/api/transactions"
   wrk.headers["Content-Type"] = "application/json"
   wrk.headers["Accept"] = "application/json"
   wrk.headers['version'] = "testnet"
   wrk.headers['port'] = 8001
   wrk.headers['nethash'] = "fl6ybowg"
   wrk.headers['os'] = ''

   body = '{"secret":"horse dinosaur brand october spoon news install tongue token pig napkin leg","amount":"%s","recipientId":"ECyhBC44N3VGop5ZkTViS5kT3AYoSwjkjo"}'
   wrk.body = string.format(body, totalCounter)

   counter = counter + 1
--    print(counter)
   return wrk.format("PUT", path)
end

-- function response(status, headers, body)
--     print(body)
-- end

File: test/credible/17.benchmark/artillery/tps.js
var node = require('../../../variables.js')

function createTransfer(address, amount, secret) {
  return node.ddn.transaction.createTransaction(address, amount, "credible", secret)
}

module.exports = {
  setJSONBody: setJSONBody,
  logHeaders: logHeaders
}

let count = 0;

function setJSONBody(req, context, ee, next) {
  var transaction = createTransfer("ECyhBC44N3VGop5ZkTViS5kT3AYoSwjkjo", 100000000 + count, node.Gaccount.password);
  count++
  // console.log(count + " transaction= ", transaction);

  req.method = "post"

  req.json = {
    transaction: transaction
  }

  req.headers = {
    'version': "testnet",
    'port': 8001,
    'nethash': "fl6ybowg",
    'os': ''
  }

  // console.log("req= ", req);

  return next(); // MUST be called for the scenario to continue
}

function logHeaders(req, response, context, ee, next) {
  console.log(response.body);
  return next(); // MUST be called for the scenario to continue
}
